src/cbc_sdk/base.py

```diff
@@ -321,20 +321,41 @@ def __str__(self):
         lines.append("-" * 79)
         lines.append("")
 
-        for attr in sorted(self._info):
-            status = "   "
-            if attr in self._dirty_attributes:
-                if self._dirty_attributes[attr] is None:
-                    status = "(+)"
-                else:
-                    status = "(*)"
-            try:
-                val = str(self._info[attr])
-            except UnicodeDecodeError:
-                val = repr(self._info[attr])
-            if len(val) > 50:
-                val = val[:47] + u"..."
-            lines.append(u"{0:s} {1:>20s}: {2:s}".format(status, attr, val))
+        try:
+            attributes = sorted(self._info)
+        except:
+            attributes = self._info
+
+        for attr in attributes:
```

Comment:
Can we add some comments here to describe the modified behavior and what the two paths are